trlx/model/accelerate_ilql_model.py

```diff
@@ -1,96 +1,56 @@
+from .accelerate_base_model import AccelerateRLModel
 
 
 @register_model
-class ILQLModel(BaseRLModel):
+class AccelerateILQLModel(AccelerateRLModel):
     def __init__(
         self,
         config,
-        tokenizer=None,
         logit_mask=None,
         train_mode=True,
     ):
         super().__init__(config, train_mode)
+        self.logit_mask = logit_mask
 
-        self.model = CausalLMWithValueHeads(
+    def get_arch(self, config):
+        return CausalLMWithValueHeads(
             config.model.model_path,
             params=config.method,
             num_layers_unfrozen=config.model.num_layers_unfrozen,
         )
-        self.max_length = config.train.gen_size
 
-        self.logit_mask = logit_mask
-        self.tokenizer = tokenizer
-
-        self.accelerator = Accelerator(log_with="wandb")
-
-        if WORLD_SIZE > 1:
-            torch.distributed.barrier(device_ids=[LOCAL_RANK])
-        else:
-            torch.random.manual_seed(1000)
-
-        if self.accelerator.is_main_process:
-            self.accelerator.init_trackers(
-                project_name=config.train.project_name, config=config.to_dict()
-            )
-
-        if self.train_mode:
-            self.opt = torch.optim.AdamW(
-                self.model.parameters(), lr=self.config.train.learning_rate_init
-            )
-            self.scheduler = rampup_decay(
-                self.config.train.lr_ramp_steps,
-                self.config.train.lr_decay_steps,
-                self.config.train.learning_rate_target
-                / self.config.train.learning_rate_init,
-                self.opt,
-            )
-
-    def tokenize(self, texts):
-        return self.tokenizer(
+    def tokenize(self, texts: Union[Iterable[str], Iterable[torch.LongTensor]]):
+        if isinstance(texts[0], torch.LongTensor):
+            return texts
+
+        tokenized = self.tokenizer(
             [self.tokenizer.bos_token + x + self.tokenizer.eos_token for x in texts],
             max_length=self.max_length,
             truncation=True,
-            padding=True,
-            return_tensors="pt",
-        )
-
-    def get_components(self) -> Dict[str, any]:
-        components = (
-            {"model": self.model, "opt": self.opt, "scheduler": self.scheduler}
-            if self.train_mode
-            else {"model": self.model}
         )
-        return components
+        input_ids = list(map(torch.as_tensor, tokenized.input_ids))
+        return input_ids
 
     def learn(self):
-        timer = Clock()
-
-        eos_token_id = self.tokenizer.eos_token_id if self.tokenizer else 0
-        train_dataloader = self.train_store.create_loader(
-            self.config.train.batch_size, eos_token_id=eos_token_id
-        )
-        eval_dataloader = self.eval_pipeline.create_loader(
-            self.config.train.batch_size, shuffle=False
-        )
+        train_dataloader = self.train_store.create_loader(self.config.train.batch_size)
```

Comment:
Where is train_store defined?

Reply:
I haven't decided on a proper way for now, so they are defined dynamically